pkg/controller/operators/olm/operator.go

```diff
@@ -406,6 +406,25 @@ func (a *Operator) handleClusterServiceVersionDeletion(obj interface{}) {
 			a.csvQueueSet.Requeue(clusterServiceVersion.GetName(), namespace)
 		}
 	}
+
+	for _, desc := range clusterServiceVersion.Spec.APIServiceDefinitions.Owned {
+		apiServiceName := fmt.Sprintf("%s.%s", desc.Version, desc.Group)
+		fetched, err := a.lister.APIRegistrationV1().APIServiceLister().Get(apiServiceName)
+		if k8serrors.IsNotFound(err) {
+			continue
+		}
+		if err != nil {
+			logger.WithError(err).Warn("api service get failure")
+		}
+		apiServiceLabels := fetched.GetLabels()
```

Comment:
Is `fetched` guaranteed to be non-nil here?

Reply:
It is. `Get(apiServiceName)` will only return nil for fetched if it is either NotFound or there is an error. The two `if` statements should handle that.

Reply:
Reading this code, a non-nil error that __is not__ `NotFound` will allow `GetLabels()` to be called.

Reply:
I see it now. It should be `continue` on the the second `if` statement I believe.

Reply:
Fixed.